logstash_async/transport.py

```diff
@@ -176,3 +227,144 @@ def send(self, events, use_logging=False):
         with client:
             for events_subset in ichunked(events, self._batch_size):
                 client.send(events_subset)
+
+
+class HttpTransport(Transport):
+    """The :class:`HttpTransport <HttpTransport>` implements a client for the
+    logstash plugin `inputs_http`.
+
+    For more details visit:
+    https://www.elastic.co/guide/en/logstash/current/plugins-inputs-http.html
+
+    :param host: The hostname of the logstash HTTP server.
+    :type host: str
+    :param port: The TCP port of the logstash HTTP server.
+    :type port: int
+    :param timeout: The connection timeout. (Default: None)
+    :type timeout: float
+    :param ssl_enable: Activates TLS. (Default: True)
+    :type ssl_enable: bool
+    :param ssl_verify: Activates the TLS certificate verification. If the flag
+    is True the class tries to verify the TLS certificate with certifi. If you
+    pass a string with a file location to CA certificate the class tries to
+    validate it against it. (Default: True)
+    :type ssl_verify: bool or str
+    :param username: Username for basic authorization. (Default: "")
+    :type username: str
+    :param password: Password for basic authorization. (Default: "")
+    :type password: str
+    :param max_content_length: The max content of an HTTP request in bytes.
+    (Default: 100MB)
+    :type max_content_length: int
+    """
+
+    def __init__(
+            self,
+            host,
+            port,
+            timeout=TimeoutNotSet,
+            ssl_enable=True,
+            ssl_verify=True,
+            **kwargs
+    ):
+        super().__init__(host, port, timeout, ssl_enable, ssl_verify)
+        self._username = kwargs.get('username', None)
+        self._password = kwargs.get('password', None)
+        self._max_content_length = kwargs.get('max_content_length', 104857600)
+        self.__session = None
+
+    @property
+    def url(self):
+        """The URL of the logstash pipeline based on the hostname, the port and
+        the TLS usage.
+
+        :return: The URL of the logstash HTTP pipeline.
+        :rtype: str
+        """
+        protocol = 'http'
+        if self._ssl_enable:
+            protocol = 'https'
+        return '{}://{}:{}'.format(protocol, self._host, self._port)
+
+    def __batches__(self, events):
```

Comment:
Sorry for being a pit picky: should be `__batches`, IMO. The trailing double underscore would make it a special method (https://docs.python.org/3/reference/datamodel.html#special-method-names) which is rather confusing.

Reply:
Sure, I change it